### PR TITLE
objects lesson

### PR DESCRIPTION
**Pull request by AnastaciaS**
objects lesson

### CODE REVIEW

File: objects.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Objects 1. BetweenDates
+
+function durationBetweenDates(startDate = new Date(), endDate = new Date(), dimension = 'seconds') {
```
Comment (sharkich) [inline objects.js]: The wrong type of default values of dates. We need to have strings here not Dates.

File: objects.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    Apples: '23.4',
+    BANANAS: '48',
+    oRAngGEs: '48.7584',
+  };
+  
+  function optimizer(priceData) {
+    const result = {};
+    for (const [key, value] of Object.entries(priceData)) {
+      result[key.toLowerCase()] = Math.round(parseFloat(value) * 100) / 100;
+    }
```
Comment (sharkich) [inline objects.js]: Looks cool. Now let's try to use methods of arrays.

File: objects.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  const result = optimizer(priceData);
+  console.log(result);
+
+
+// Objects 3. Recursive
+
+function recursiveOddSumTo(number) {
+    if (number <= 0) {
+      return 0;
+    } else if (number % 2 === 0) {
```
Comment (sharkich) [inline objects.js]: No need for `else` if you have the `return` inside the previous statement.

**Review by sharkich — CHANGES_REQUESTED** [review]

File: objects.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const startTime = Date.parse(startDate);
+    const endTime = Date.parse(endDate);
+    
+    const diff = Math.abs(startTime - endTime);
+    const result = Math.floor(diff / units[dimension]);
+    
+    return `${result} ${dimension}`;
+  }
+  
+  console.log(durationBetweenDates('21 Jul 2025', '19 Mar 2022', 'seconds'));
```
Comment (sharkich) [inline objects.js]: Screenshot 2023-03-24 at 12 00 18

NaN is wrong answer. Something wrong with default values

**Review by sharkich — CHANGES_REQUESTED** [review]

File: objects.js (hunk 4 above)
Comment (AnastaciaS) [reply objects.js]: Hmm, but you have no dates in the console log. If you specify the dates, everything will work, won't it? Or have I misunderstood something in the task?

File: objects.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+`use strict`;
+// Objects 1. BetweenDates
+
+function durationBetweenDates(startDate = 'now', endDate = 'now', dimension = 'seconds') {
```
Comment (sharkich) [inline objects.js]: default value as a string of date

File: objects.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+
+// Objects 3. Recursive
+
+function recursiveOddSumTo(number) {
+    if (number <= 0) {
+      return 0;
+    } if (number % 2 === 0) {
+      return recursiveOddSumTo(number - 1);
+    } {
```
Comment (sharkich) [inline objects.js]: unnessesery block

**Review by sharkich — CHANGES_REQUESTED** [review]

**Review by sharkich — APPROVED** [review]

**Pull request merged**